src/js/component/text.js

```diff
@@ -282,14 +282,23 @@ class Text extends Component {
      */
     setStyle(activeObj, styleObj) {
         return new Promise(resolve => {
-            snippet.forEach(styleObj, (val, key) => {
-                if (activeObj[key] === val) {
-                    styleObj[key] = resetStyles[key] || '';
-                }
-            }, this);
-
-            activeObj.set(styleObj);
-
+            const activeObject = this.getCanvas().getActiveObject();
+            if (styleObj.textAlign || activeObject.selectionEnd === activeObject.selectionStart) {
+                snippet.forEach(styleObj, (val, key) => {
+                    if (activeObj[key] === val) {
+                        styleObj[key] = resetStyles[key] || '';
+                    }
+                }, this);
+                activeObj.set(styleObj);
+            } else {
+                snippet.forEach(styleObj, (val, key) => {
+                    const selectedPart = activeObj.getSelectionStyles();
+                    if (selectedPart[key] === val) {
+                        styleObj[key] = resetStyles[key] || '';
+                    }
+                }, this);
+                activeObject.setSelectionStyles(styleObj);
+            }
```

Comment:
I think this `forEach` loop could be extracted and used once. It has duplication code.

How about this?

```js
//...
            const selectionLength = activeObj.selectionEnd - activeObj.selectionStart;
            const isSelectionCollapsed = selectionLength === 0;
            let targetObj = activeObj;

            if (!styleObj.textAlign && !isSelectionCollapsed) {
                targetObj = activeObj.getSelectionStyles();
            }

            snippet.forEach(styleObj, (val, key) => {
                if (targetObj[key] === val) {
                    styleObj[key] = resetStyles[key] || '';
                }
            }, this);

            if (isSelectionCollapsed) {
                activeObj.set(styleObj);
            } else {
                activeObj.setSelectionStyles(styleObj);
            }
// ...
```

Reply:
Refactored code and remove duplicate code. Also fix issue, if no character selected, before apply style it will select all characters and apply style for them. 

But now, some tests are failing. They check whole object style, not character style. Now i am trying to figure out, how to fix tests.